// @gokiprotocol/walletkit/dist/esm/components/WalletSelectorModal/WalletStepSelect/WalletProviderOption.js
import { jsx as _jsx, jsxs as _jsxs } from "@emotion/react/jsx-runtime";
import styled from "@emotion/styled";
import { useMemo } from "react";
export const WalletProviderOption = ({ type, info, onInstall, onSelect, }) => {
    var _a;
    const mustInstall = typeof window !== "undefined" && ((_a = info.isInstalled) === null || _a === void 0 ? void 0 : _a.call(info)) === false;
    const icon = typeof info.icon === "string" ? (_jsx("img", { src: info.icon, alt: `Icon for wallet ${info.name}` })) : (_jsx(info.icon, {}));
    const providerURL = useMemo(() => {
        try {
            const name = new URL(info.url).hostname;
            if (name.startsWith("www.")) {
                return name.slice(4);
            }
            return name;
        }
        catch (e) {
            return info.url;
        }
    }, [info.url]);
    return (_jsx(Wrapper, { id: `gokiWalletKitProviderOption-${type}`, role: "button", onClick: (e) => {
            e.stopPropagation();
            e.preventDefault();
            if (mustInstall) {
                onInstall === null || onInstall === void 0 ? void 0 : onInstall(info);
                return;
            }
            onSelect === null || onSelect === void 0 ? void 0 : onSelect();
        }, children: _jsx(InfoTileWrapper, { className: "wallet-info-tile", children: _jsxs(InfoTile, { children: [_jsx(IconWrapper, { children: icon }), _jsxs(ProviderDesc, { children: [_jsx(ProviderName, { children: info.name }), _jsxs(ProviderUrl, { children: [providerURL, mustInstall ? " (not installed)" : ""] })] })] }) }) }));
};
const IconWrapper = styled.div `
  height: 33px;
  width: 33px;

  & > img,
  & > svg {
    height: 100%;
    width: 100%;
  }
`;
const InfoTileWrapper = styled.div `
  flex: 1 1 auto;
  height: 100%;

  display: flex;
  align-items: center;
`;
const InfoTile = styled.div `
  display: grid;
  grid-template-columns: 33px 1fr;
  grid-column-gap: 16px;
`;
const ProviderDesc = styled.div `
  display: flex;
  flex-direction: column;
`;
const ProviderName = styled.span `
  font-weight: 600;
  font-size: 14px;
  line-height: 18px;
  letter-spacing: -0.02em;
  color: #000000;
`;
const ProviderUrl = styled.span `
  font-weight: normal;
  font-size: 12px;
  line-height: 15px;
  letter-spacing: -0.02em;
  color: #696969;
`;
const Wrapper = styled.div `
  width: 100%;
  height: 65px;
  user-select: none;
  cursor: pointer;
  padding: 0 28px;

  display: flex;
  align-items: center;

  background: #fff;
  .wallet-info-tile {
    border-bottom: 1px solid #f8f8f8;
  }
  &:hover {
    background: #f9f9f9;
    .wallet-info-tile {
      border-bottom: 1px solid #e6e6e6;
    }
  }
`;
