// @gokiprotocol/walletkit/dist/cjs/components/WalletSelectorModal/WalletStepSecretKey/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WalletStepSecretKey = void 0;
const tslib_1 = require("tslib");
const jsx_runtime_1 = require("@emotion/react/jsx-runtime");
/* eslint-disable jsx-a11y/anchor-is-valid */
const react_1 = require("@emotion/react");
const styled_1 = tslib_1.__importDefault(require("@emotion/styled"));
const use_solana_1 = require("@saberhq/use-solana");
const web3_js_1 = require("@solana/web3.js");
const react_2 = require("react");
const LabeledInput_1 = require("../../LabeledInput");
const ButtonWithFooter_1 = require("../ButtonWithFooter");
const WalletStepSecretKey = ({ onBack, onSuccess, onError, }) => {
    const [keypairStr, setKeypairStr] = (0, react_2.useState)(JSON.stringify([...web3_js_1.Keypair.generate().secretKey]));
    const [keypair, keypairErr] = (0, react_2.useMemo)(() => {
        try {
            return [
                web3_js_1.Keypair.fromSecretKey(Uint8Array.from(JSON.parse(keypairStr))),
                null,
            ];
        }
        catch (e) {
            return [null, e];
        }
    }, [keypairStr]);
    const { activate } = (0, use_solana_1.useSolana)();
    return ((0, jsx_runtime_1.jsxs)(Wrapper, { children: [(0, jsx_runtime_1.jsx)(IconWrapper, { children: (0, jsx_runtime_1.jsx)(use_solana_1.DEFAULT_WALLET_PROVIDERS.SecretKey.icon, {}) }), (0, jsx_runtime_1.jsx)("h2", { children: "Enter a JSON Keypair" }), (0, jsx_runtime_1.jsx)("p", { css: (0, react_1.css) `
          color: red !important;
        `, children: "Warning: do not use this outside of testing. If you were told to go here by someone else, don't do it." }), (0, jsx_runtime_1.jsxs)(Fields, { children: [(0, jsx_runtime_1.jsx)(LabeledInput_1.LabeledInput, { label: "Keypair (JSON)", placeholder: "Enter a JSON keypair string", name: "account", value: keypairStr, onChange: (e) => {
                            setKeypairStr(e.target.value);
                        } }), keypair && ((0, jsx_runtime_1.jsx)(LabeledInput_1.LabeledInput, { id: "secretKeyConnectorPubkey", label: "Public Key", name: "publicKey", disabled: true, value: keypair === null || keypair === void 0 ? void 0 : keypair.publicKey.toString() }))] }), keypairErr && ((0, jsx_runtime_1.jsx)("p", { children: (0, jsx_runtime_1.jsxs)("span", { css: (0, react_1.css) `
              color: red;
            `, children: ["Error: ", keypairErr.message] }) })), (0, jsx_runtime_1.jsx)(ButtonWithFooter_1.ButtonWithFooter, { id: "walletKit_secretKey_continue", disabled: !keypair, 
                // eslint-disable-next-line @typescript-eslint/no-misused-promises
                onClick: async () => {
                    if (!keypair) {
                        throw new Error("keypair missing");
                    }
                    try {
                        await activate(use_solana_1.DefaultWalletType.SecretKey, {
                            secretKey: [...keypair.secretKey],
                        });
                    }
                    catch (e) {
                        onError === null || onError === void 0 ? void 0 : onError(e);
                        return;
                    }
                    onSuccess === null || onSuccess === void 0 ? void 0 : onSuccess();
                }, footer: (0, jsx_runtime_1.jsxs)(jsx_runtime_1.Fragment, { children: ["Having trouble?", " ", (0, jsx_runtime_1.jsx)("a", { href: "#", onClick: (e) => {
                                e.preventDefault();
                                e.stopPropagation();
                                onBack === null || onBack === void 0 ? void 0 : onBack();
                            }, children: "Go back" })] }), children: "Continue" })] }));
};
exports.WalletStepSecretKey = WalletStepSecretKey;
const IconWrapper = styled_1.default.div `
  & > svg,
  & > img {
    width: 36px;
    height: 36px;
  }
  margin-bottom: 32px;
`;
const Wrapper = styled_1.default.div `
  padding: 28px;
  padding-top: 67px;

  & > h2 {
    font-weight: bold;
    font-size: 20px;
    line-height: 25px;
    letter-spacing: -0.02em;
    color: #000000;
  }

  & > p {
    font-weight: normal;
    font-size: 14px;
    line-height: 18px;
    letter-spacing: -0.02em;
    color: #696969;
  }
`;
const Fields = styled_1.default.div `
  display: flex;
  flex-direction: column;
  gap: 8px;
  width: 100%;
`;
